libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c

```diff
@@ -165,6 +165,11 @@ static IotMqttNetworkInfo_t _networkInfo = IOT_MQTT_NETWORK_INFO_INITIALIZER;
  */
 static IotNetworkInterface_t _networkInterface = { 0 };
 
+/**
+ * @brief A packet allocated by _serializePingreq.
+ */
+static uint8_t * _pAllocatedPingreq = NULL;
```

Comment:
Why does this need to be global?

Reply:
It's shared between _serializePingreq and the test function.  It's set by _serializePingreq so the test function knows what to free to clean up.